lib/services/plugins-service.ts

```diff
@@ -178,32 +216,33 @@ export class PluginsService implements IPluginsService {
 	private createPluginsData(pluginsService: ICordovaPluginsService): IFuture<void> {
 		return (() => {
 			var plugins = pluginsService.getAvailablePlugins().wait();
-
 			_.each(plugins, (plugin: Server.CordovaPluginData) => {
-
 				try {
-					var pluginData = pluginsService.createPluginData(plugin).wait();
-				} catch(e) {
+					var data = pluginsService.createPluginData(plugin);
+				} catch (e) {
 					this.$logger.warn("Unable to fetch data for %s plugin. Please, try again in a few minutes.", (<any>plugin).title);
 					this.$logger.trace(e);
 				}
-
-				if(pluginData) {
-					var projectDataRecord = pluginData.toProjectDataRecord();
-					var configurations = _.keys(this.$project.configurationSpecificData);
-
-					_.each(configurations, (configuration:string) => {
-						var configData = this.$project.configurationSpecificData[configuration];
-						if (configData) {
-							var corePlugins = configData[PluginsService.CORE_PLUGINS_PROPERTY_NAME];
-							if (corePlugins && _.contains(corePlugins, projectDataRecord)) {
-								pluginData.configurations.push(configuration);
+				_.each(data, pluginData => {
+					if (pluginData && pluginData.data) {
+						var projectDataRecord = pluginData.toProjectDataRecord();
+						var configurations = _.keys(this.$project.configurationSpecificData);
```

Comment:
Maybe @teobugslayer  was talking about next each, where you are `eaching` (I love this new word :smiley_cat: ) configurations and you are also using 
```TypeScript
var configData = this.$project.configurationSpecificData[configuration];
```
The idea is to take all configurations here (where @teobugslayer  commented), not only the keys, as it looks like you need both keys and values.